### PR TITLE
Python: Fix date/time transforms

### PR DESCRIPTION
**Pull request by rdblue**
This fixes some of the date/time transform methods introduced in #5462.

### CODE REVIEW

File: python/pyiceberg/utils/datetime.py [hunk 1]
```diff
@@ -133,28 +133,24 @@ def to_human_timestamp(timestamp_micros: int) -> str:
...
 
 
 def days_to_years(days: int) -> int:
     return days_to_date(days).year - EPOCH_DATE.year
 
 
-def micros_to_years(timestamp: int) -> int:
-    dt = micros_to_timestamp(timestamp)
-    return (dt.year - EPOCH_TIMESTAMP.year) - (
-        1 if dt.month < EPOCH_TIMESTAMP.month or (dt.month == EPOCH_TIMESTAMP.month and dt.day < EPOCH_TIMESTAMP.day) else 0
```
Comment (rdblue) [inline python/pyiceberg/utils/datetime.py:159]: Month and day are never 0.

File: python/pyiceberg/utils/datetime.py [hunk 2]
```diff
@@ -133,28 +133,24 @@ def to_human_timestamp(timestamp_micros: int) -> str:
     return (EPOCH_TIMESTAMP + timedelta(microseconds=timestamp_micros)).isoformat()
 
 
-def micros_to_hours(timestamp: int) -> int:
-    """Converts a timestamp in microseconds to a date in hours"""
-    return int((datetime.utcfromtimestamp(timestamp // 1_000_000) - EPOCH_TIMESTAMP).total_seconds() / 3600)
+def micros_to_hours(micros: int) -> int:
+    """Converts a timestamp in microseconds to hours from 1970-01-01T00:00"""
+    return micros // 3_600_000_000
```
Comment (rdblue) [inline python/pyiceberg/utils/datetime.py:138]: This converts to a timestamp only to get back to a delta, and then get the seconds. Instead, we can convert directly.

File: python/pyiceberg/utils/datetime.py [hunk 3]
```diff
@@ -35,7 +35,7 @@
 
 def micros_to_days(timestamp: int) -> int:
     """Converts a timestamp in microseconds to a date in days"""
-    return (datetime.fromtimestamp(timestamp / 1_000_000) - EPOCH_TIMESTAMP).days
+    return timedelta(microseconds=timestamp).days
```
Comment (rdblue) [inline python/pyiceberg/utils/datetime.py:38]: Here, we can go directly to delta rather than going to date and subtracting.

**Review by Fokko — APPROVED** [review]
Looks good to me, thanks @rdblue 

Also, this is already properly tested:
```
poetry run coverage report -m --fail-under=90
Name                                       Stmts   Miss  Cover   Missing
------------------------------------------------------------------------
...
pyiceberg/utils/datetime.py                   69      0   100% 
```

**Pull request merged**